src/components/text-editor/__internal__/plugins/Toolbar/buttons/list.component.tsx

```diff
@@ -0,0 +1,377 @@
+      listNode = listNode.getParent();
+    }
+
+    /* istanbul ignore else */
+    if (listNode instanceof ListNode) {
+      setIsOLActive(listNode.getListType() === "number");
+      setIsULActive(listNode.getListType() === "bullet");
+    } else {
+      setIsOLActive(false);
+      setIsULActive(false);
+    }
+  }, []);
+
+  // Register an update listener to update the toolbar when the editor state changes
+  useEffect(() => {
+    return mergeRegister(
+      editor.registerUpdateListener(({ editorState }) => {
+        editorState.read(() => {
+          const isEditable = editor.isEditable();
+          /* istanbul ignore else */
+          if (isEditable) updateToolbar();
+        });
+      }),
+    );
+  }, [updateToolbar, editor]);
+
+  // When the ordered list button is clicked, insert or remove an ordered list
+  const handleOLClick = () => {
+    const isEditable = editor.isEditable();
+    /* istanbul ignore if */
+    if (!isEditable) return;
+
+    // Focus the editor before dispatching the command
+    editor.focus();
+
+    if (isOLActive) {
+      editor.dispatchCommand(REMOVE_LIST_COMMAND, undefined);
+      setIsOLActive(false);
+    } else {
+      editor.dispatchCommand(INSERT_ORDERED_LIST_COMMAND, undefined);
+      setIsOLActive(true);
+    }
+    // If the unordered list button is active, deactivate it
+    setIsULActive(false);
+  };
+
+  // When the unordered list button is clicked, insert or remove an unordered list
+  const handleULClick = () => {
+    const isEditable = editor.isEditable();
+    /* istanbul ignore if */
+    if (!isEditable) return;
+
+    // Focus the editor before dispatching the command
+    editor.focus();
+
+    if (isULActive) {
+      editor.dispatchCommand(REMOVE_LIST_COMMAND, undefined);
+      setIsULActive(false);
+    } else {
+      editor.dispatchCommand(INSERT_UNORDERED_LIST_COMMAND, undefined);
+      setIsULActive(true);
+    }
+    // If the ordered list button is active, deactivate it
+    setIsOLActive(false);
+  };
+
+  // Split an array into three parts: before a subset, the subset, and after the subset
+  function splitArray(allEntries: string[], subsetEntries: string[]) {
+    let startIndex = -1;
+    for (let i = 0; i <= allEntries.length - subsetEntries.length; i++) {
+      if (
+        allEntries.slice(i, i + subsetEntries.length).toString() ===
+        subsetEntries.toString()
+      ) {
+        startIndex = i;
+        break;
+      }
+    }
+
+    /* istanbul ignore if */
+    if (startIndex === -1) {
+      throw new Error("Subset B not found in array A");
+    }
+
+    // If the subset matches the full selection, return the original subset
+    if (startIndex === 0 && subsetEntries.length === allEntries.length) {
+      return { beforeSubset: [], subset: allEntries, afterSubset: [] };
+    }
+
+    // Split A into three parts
+    const beforeSubset = allEntries.slice(0, startIndex);
+    const subset = allEntries.slice(
```

Comment:
question: is `subsetEntries` not the subset or is there a potential edge case slicing from `allEntries` covers here?